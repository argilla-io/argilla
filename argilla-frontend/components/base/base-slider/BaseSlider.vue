<!--
  - coding=utf-8
  - Copyright 2021-present, the Recognai S.L. team.
  -
  - Licensed under the Apache License, Version 2.0 (the "License");
  - you may not use this file except in compliance with the License.
  - You may obtain a copy of the License at
  -
  -     http://www.apache.org/licenses/LICENSE-2.0
  -
  - Unless required by applicable law or agreed to in writing, software
  - distributed under the License is distributed on an "AS IS" BASIS,
  - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  - See the License for the specific language governing permissions and
  - limitations under the License.
  -->

<template>
  <div class="slides-nav">
    <a
      :class="itemNumber <= 0 ? 'disabled' : null"
      href="#"
      @click.prevent="prev(itemNumber)"
    >
      <svgicon name="chevron-left" width="8" height="8" color="#4C4EA3" />
    </a>
    {{ itemNumber + 1 }} of {{ slidesOrigin.length }} {{ slidesName }}
    <a
      :class="slidesOrigin.length <= itemNumber + 1 ? 'disabled' : null"
      href="#"
      @click.prevent="next(itemNumber)"
    >
      <svgicon name="chevron-right" width="8" height="8" />
    </a>
  </div>
</template>

<script>
export default {
  props: {
    itemNumber: {
      type: Number,
      required: true,
    },
    slidesName: {
      type: String,
    },
    slidesOrigin: {
      type: Array,
      required: true,
    },
  },
  methods: {
    prev(number) {
      this.$emit("go-to", --number);
    },
    next(number) {
      this.$emit("go-to", ++number);
    },
  },
};
</script>

<style scoped lang="scss">
.slides-nav {
  @include font-size(13px);
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 33%;
  margin-right: 33%;
  margin-left: auto;
  color: $black-54;
  a {
    height: 20px;
    width: 20px;
    line-height: 19px;
    border-radius: $border-radius-s;
    text-align: center;
    margin-left: 1.5em;
    margin-right: 1.5em;
    display: inline-block;
    text-decoration: none;
    outline: none;
    @include font-size(13px);
    transition: all 0.2s ease-in-out;
    background: palette(grey, 800);
    .svg-icon {
      fill: $black-87;
    }
    &:hover {
      background: palette(grey, 700);
      transition: all 0.2s ease-in-out;
    }
    &.disabled {
      opacity: 0.5;
      pointer-events: none;
    }
  }
}
</style>
