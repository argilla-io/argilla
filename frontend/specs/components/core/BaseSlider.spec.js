import { mount } from "@vue/test-utils";
import BaseSlider from "@/components/base/BaseSlider";

function mountBaseSlider() {
  return mount(BaseSlider, {
    propsData: {
      slidesName: "sentences",
      slidesOrigin: ["first sentence", "second sentence"],
      itemNumber: 0,
    },
  });
}

describe("BaseSlider", () => {
  const spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("renders properly", () => {
    const wrapper = mountBaseSlider();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
