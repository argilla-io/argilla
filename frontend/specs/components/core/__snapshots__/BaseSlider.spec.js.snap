// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSlider renders properly 1`] = `
<div class="slides-nav"><a href="#" class="disabled"><svg version="1.1" viewBox="0 0 8 8" class="svg-icon svg-fill" style="width: 8px; height: 8px;"></svg></a>
  1 of 2 sentences
  <a href="#"><svg version="1.1" viewBox="0 0 8 8" class="svg-icon svg-fill" style="width: 8px; height: 8px;"></svg></a>
</div>
`;
